File: backend-asp-school/backend-asp-school/Models/Subject.cs
using System.Text.Json.Serialization;

namespace backend_asp_school.Models
{
    public class Subject
    {
        public int SubjectID { get; set; }
        public string SubjectName { get; set; } = null!;
        public string? Description { get; set; }
        [JsonIgnore]
        public ICollection<Result>? Results { get; set; }
        public ICollection<Lesson>? Lessons { get; set; }
        public ICollection<Exam>? Exams { get; set; }
    }
}

File: next-dashboard-ui/src/lib/subject.ts
import { Result } from "./result";
import { Lesson } from "./lesson";
import { Exam } from "./exam";

export interface Subject {
  subjectID: number;
  subjectName: string;
  description?: string;

  results?: Result[];
  lessons?: Lesson[];
  exams?: Exam[];
}
